bootloader/core/util.c

```diff
@@ -78,3 +78,43 @@ uint16_t compute_checksum(const uint8_t * begin, const uint8_t * end) {
     }
     return (~computed_checksum + 1) & 0xFFFF;
 }
+
+
+
+/**
+ * Iterate a byte buffer creating double word values and addresses.
+ *
+ * Writes to G4 and L5 flash happen in 64 bit data increments.
+ * @param address The starting flash address
+ * @param buffer A byte buffer
+ * @param length Length in bytes of the buffer.
+ * @param callback A callback that will be called for each 64 bit address
+ * @return True on success
+ */
+bool dword_address_iter(uint32_t address, const uint8_t * buffer, uint8_t length, address_iter_callback callback) {
+    if (!buffer || length <= 0) {
```

Comment:
What does `!buffer` mean in this context. Is this a conditional like `not variable` in python?
If so are you just verifying that it is not 0?

Reply:
Is there a reason that you don't use `!=0`?

Reply:
idiomatic